services/app/src/contexts/ChessBoardContext.jsx

```diff
@@ -0,0 +1,47 @@
+/* eslint-disable react/jsx-no-constructed-context-values */
+/* eslint-disable react/prop-types */
+import React, { createContext, useState } from 'react';
+import { useChessBoard } from '../hooks/useChessBoard';
+/* eslint-disable-next-line */
+import { calculateTheScore } from '../hooks/calculateResult';
+import '../static/style/gameScore.css';
+
+export const ChessBoardContext = createContext();
+let num;
+
+const ChessBoardContextProvider = ({ children }) => {
+  const [selectedPiece, setSelectedPiece] = useState('');
+  const [board, setBoard] = useChessBoard();
+
+  const placeItemOnBoard = (row, column) => {
+    if (selectedPiece) {
+      const copyOfBoard = [...board];
+      copyOfBoard[row][column].items.push(selectedPiece);
```

Comment:
Updating the board state should happen in backend. As well as score calculation. We should do minimal business logic in front end. We already have WS endpoints for board state. Not sure if we have score calculation, but it should happen in backend.